JitsiConference.js

```diff
@@ -2054,20 +2055,38 @@ JitsiConference.prototype._fireIncompatibleVersionsEvent = function() {
  * @param to {string} the id of the endpoint that should receive the message.
  * If "" the message will be sent to all participants.
  * @param payload {object} the payload of the message.
+ * @param sendThroughVideobridge {boolean} whether to use the jitsi-videobridge
+ * based transport (if true) or the MUC based transport (if false).
  * @throws NetworkError or InvalidStateError or Error if the operation fails.
  */
-JitsiConference.prototype.sendEndpointMessage = function(to, payload) {
-    this.rtc.sendChannelMessage(to, payload);
-};
+JitsiConference.prototype.sendEndpointMessage
+    = function(to, payload, sendThroughVideobridge = true) {
+        if (sendThroughVideobridge) {
+            this.rtc.sendChannelMessage(to, payload);
+        } else {
+            const message = { payload };
+
+            message[JITSI_MEET_MUC_TOPIC] = '';
+
+            if (to === '') {
+                this.sendTextMessage(message);
+            } else {
+                this.sendPrivateTextMessage(to, message);
+            }
+        }
+    };
```

Comment:
I want your thoughts only because the change set still looks okay to me right now:
What about having new methods for sending a private message and a public message? My idea being that functions should try to do one thing only, and generally a boolean used to do different behaviors is a sign maybe multiple functions are warranted. With separate functions we know sendEndpointMessage always goes through the bridge data/websock channel.

Reply:
Thanks for bringing this up. This was written for shipit, so I didn't really think about it at the time.

I am concerned about this interface getting too complex and inconsistent, but I don't know what the best way out is. There are 8 distinct cases, according to:
a. Whether the message is private or public
b. Whether we send it over the bridge channel or over the MUC
c. Whether we are passing a string or a JSON object (to be encoded in a special way)

IMO 8 or 4 separate methods is too much, so I'd like to bring it down to 2. Perhaps we can handle "c" according to the type of the argument. And between "a" and "b" I would choose to have different methods for private vs public, which leaves using an argument for "b".

One argument for using the same function to send over the bridge and over the MUC is that on the receive side we don't really care which channel the message came from, we just fire the same event in both cases. I think that's a good design, because it masks some of the complexity. Also, it's easy to chose a reasonable default (either one is reasonable, IMO).

So I would suggest:
```
sendMessage(message, sendThroughVideobridge = true)
sendPrivateMessage(message, sendThroughVideobridge = true)
```
Where `message` is either a string or an object and objects are encoded as needed.

We can then deprecate the rest of the methods. What do you think?

Reply:
What's the use case for setting `sendThroughVideobridge` to true versus false? I realized that I don't know for what reasons true or false would be set. Maybe that's why my desire was to make the branching point for functions on `sendThroughVideobridge`, and your explanation has revealed that ignorance to me. That said, splitting on public/private also makes logical sense to me, and maybe more sense from an api consumer standpoint, but I definitely don't have context to say anything more than "sounds reasonable."

Reply:
Sending through the bridge will make the message available to the bridge but not jicofo (and not jigasi, today), sending through the MUC makes it available to jicofo and jigasi but not the bridge. It's very technical, which is why I'd like it to be somewhat hidden.

Reply:
Okay. I'm still not sure of why you need to send some something to the bridge but not jicofo/jigasi. Do you have an example, either real or hypothetical? Would there be a time when you need to send to all?